src/github.com/getlantern/eventual/eventual.go

```diff
@@ -51,34 +56,59 @@ func DefaultGetter(val interface{}) Getter {
 }
 
 func (v *value) Set(val interface{}) {
-	v.updates <- val
+	// Prevent sending on closed channel
+	if atomic.LoadInt32(&v.stopped) == intFalse {
+		v.muUpdates.RLock()
+		v.updates <- val
+		v.muUpdates.RUnlock()
+	}
 }
 
 func (v *value) processUpdates() {
 	for val := range v.updates {
 		v.val.Store(val)
-		if v.gotFirst == FALSE {
+		if v.gotFirst == intFalse {
```

Comment:
To be safe, we should probably use atom.LoadInt32(&v.gotFirst).  In fact, we could just make a new utility function:

```go
func (v *value) signalFirst() {
    if atomic.CompareAndSwapInt32(&v.gotFirst, intFalse, intTrue) {
        v.wg.Done()
    }
}
```

And then use that both inside the loop and after the loop.